src/Aspire.Cli/Certificates/CertificateService.cs

```diff
@@ -48,22 +48,20 @@ public async Task EnsureCertificatesTrustedAsync(IDotNetCliRunner runner, Cancel
                     options,
                     cancellationToken));
 
-            if (trustExitCode != 0)
-            {
-                var outputLines = ensureCertificateCollector.GetLines();
-
-                if (outputLines.Any(line => line.Line == DevCertsPartialTrustMessage))
-                {
-                    // On some platforms the trust command may return with a non-zero exit code by still
-                    // be functional enough to work for .NET Aspire. This is a workaround for that non-zero
-                    // exit code that allows the CLI to continue starting up the apphost. We want to warn
-                    // when this happens so we know we are hitting this corner case.
-                    interactionService.DisplayMessage(
-                        "warning",
-                        "The HTTPS developer certificate is partially trusted. Some clients may not work correctly.");
-                    return;
-                }
+            var outputLines = ensureCertificateCollector.GetLines();
 
+            // Exitcode 4 means that the certs were not trusted, but there is a condition where they
+            // are partially trusted which we will treat as not fatal, but we will show a warning for
+            // diagnostic purposes just in case it is a problem for the user.
+            if (trustExitCode == 4 && outputLines.Any(line => line.Line == DevCertsPartialTrustMessage))
+            {
+                interactionService.DisplayMessage(
+                    "warning",
+                    "The HTTPS developer certificate is partially trusted. Some clients may not work correctly.");
+                return;
+            }
+            else if (trustExitCode != 0)
```

Comment:
We need to ask ourselves if it's OK that we stop you from running wholesale. I would argue that a non-zero exit code should just warn and continue running.

Reply:
Fair comment. I don't have a strong opinion. Depending on the failure here its highly likely that nothing will work in terms of service to service comms and the dashboard. So this is mostly about best effort to make it functional. But it doesn't need to fatally block you.